sympy/core/relational.py

```diff
@@ -507,6 +507,8 @@ def expand(self, **kwargs):
         return self.func(*args)
 
     def __bool__(self):
+        if self.lhs.equals(self.rhs):
```

Comment:
`equals` can be very expensive. It is perhaps better to use `self.lhs == self.rhs`.

Reply:
Thank you for the review! As far as I understand, these test different things. We need ``equals()`` because for this fix, we explicitly need to check equivalence rather than precisely the same syntax. In the provided test case, for example:

```py
(Pdb) self.lhs
1.0*a*c - 1.0*b*c
(Pdb) self.rhs
a*c - b*c
(Pdb) self.lhs == self.rhs
False
(Pdb) self.lhs.equals(self.rhs)
True
```